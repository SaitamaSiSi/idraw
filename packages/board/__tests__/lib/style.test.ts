import {
  mergeCSS2StyleAttr, setStyle, getStyle, getDomTransform,
  setDomTransform,
} from './../../src/lib/style';


describe('idraw_zyh_board: src/lib/style', () => {
  test('mergeCSS2StyleAttr', () => {
    const str = mergeCSS2StyleAttr({
      'width': '12px',
      'height': '20px',
      'overflow': 'auto',
      'margin-top': '123px',
    });
    expect(str).toStrictEqual('width:12px; height:20px; overflow:auto; margin-top:123px;');
  });

  test('setStyle', () => {
    const div = document.createElement('div');
    setStyle(div, {
      'width': '12px',
      'height': '20px',
      'overflow': 'auto',
      'margin-top': '123px',
    });
    expect(div).toMatchSnapshot();
  });

  test('getStyle', () => {
    const div = document.createElement('div');
    setStyle(div, {
      'width': '12px',
      'height': '20px',
      'overflow': 'auto',
      'margin-top': '123px',
    });
    expect(getStyle(div)).toMatchSnapshot({
      'width': '12px',
      'height': '20px',
      'overflow': 'auto',
      'margin-top': '123px',
    });
  });

  test('setDomTransform', () => {
    // transform: matrix( scaleX(), skewY(), skewX(), scaleY(), translateX(), translateY() )
    // matrix(1, 2, -1, 1, 80, 80)
    const div = document.createElement('div');
    setDomTransform(div, {
      scaleX: 1,
      skewY: 2,
      skewX: -1,
      scaleY: 1,
      translateX: 80,
      translateY: 90,
    });
    expect(div).toMatchSnapshot();
  })

  test('getDomTransform', () => {
    // transform: matrix( scaleX(), skewY(), skewX(), scaleY(), translateX(), translateY() )
    // matrix(1, 2, -1, 1, 80, 80)
    const div = document.createElement('div');
    setDomTransform(div, {
      scaleX: 1,
      skewY: 2,
      skewX: -1,
      scaleY: 1,
      translateX: 80,
      translateY: 90,
    });
    expect(getDomTransform(div)).toStrictEqual({
      scaleX: 1,
      skewY: 2,
      skewX: -1,
      scaleY: 1,
      translateX: 80,
      translateY: 90,
    });
  })
})