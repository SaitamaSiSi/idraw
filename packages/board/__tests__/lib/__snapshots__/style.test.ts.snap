// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@idraw/board: src/lib/style getStyle 1`] = `
Object {
  "height": "20px",
  "margin-top": "123px",
  "overflow": "auto",
  "width": "12px",
}
`;

exports[`@idraw/board: src/lib/style setDomTransform 1`] = `
<div
  style="transform: matrix(1, 2, -1, 1, 80, 90);"
/>
`;

exports[`@idraw/board: src/lib/style setStyle 1`] = `
<div
  style="width:12px;height:20px;overflow:auto;margin-top:123px;"
/>
`;
